### ISSUE TITLE
installation problem with MKL forcing most recent pangeo-pytide version

### ISSUE THREAD

**Issue by jerabaul29** [CNES/pangeo-pytide#7]
I work on Ubuntu 20.04, all packages up to date as far as I know.

```
(base) ~$ conda --version
conda 4.8.5
```

I am having some issues with installation of the most recent pangeo-pytide version. MKS seems to not work. Installing in the base env "automagically" solved this by downgrading to v 1.2.0, but this has issues of its own (see issue #5 ).

Basically, I have tried both to install MKS by hand first, or to let conda take care of everything, but in both case, I cannot run the tests on the latest 1.2.2:

```
(my_pytide) ~/Desktop/Git/pangeo-pytide [master|✔]> conda install mkl-devel
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/jrmet/anaconda3/envs/my_pytide

  added / updated specs:
    - mkl-devel


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    intel-openmp-2020.2        |              254         786 KB
    mkl-2020.2                 |              256       138.3 MB
    mkl-devel-2020.2           |              257          14 KB
    ------------------------------------------------------------
                                           Total:       139.1 MB

The following NEW packages will be INSTALLED:

  blas               pkgs/main/linux-64::blas-1.0-mkl
  intel-openmp       pkgs/main/linux-64::intel-openmp-2020.2-254
  mkl                pkgs/main/linux-64::mkl-2020.2-256
  mkl-devel          pkgs/main/linux-64::mkl-devel-2020.2-257
  mkl-include        pkgs/main/linux-64::mkl-include-2020.2-256


Proceed ([y]/n)? y


Downloading and Extracting Packages
intel-openmp-2020.2  | 786 KB    | ################################################################################################################################################################### | 100% 
mkl-devel-2020.2     | 14 KB     | ################################################################################################################################################################### | 100% 
mkl-2020.2           | 138.3 MB  | ################################################################################################################################################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
(my_pytide) ~/Desktop/Git/pangeo-pytide [master|✔]> conda install pytide=1.2.2 -c conda-forge
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/jrmet/anaconda3/envs/my_pytide

  added / updated specs:
    - pytide=1.2.2


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    mkl-service-2.3.0          |   py38h516909a_0          67 KB  conda-forge
    mkl_fft-1.1.0              |   py38hc1659b7_1         180 KB  conda-forge
    mkl_random-1.1.0           |   py38hb3f55d8_0         394 KB  conda-forge
    six-1.15.0                 |     pyh9f0ad1d_0          14 KB  conda-forge
    ------------------------------------------------------------
                                           Total:         654 KB

The following NEW packages will be INSTALLED:

  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge
  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-1_llvm
  ca-certificates    conda-forge/linux-64::ca-certificates-2020.6.20-hecda079_0
  certifi            conda-forge/linux-64::certifi-2020.6.20-py38h32f6830_0
  ld_impl_linux-64   conda-forge/linux-64::ld_impl_linux-64-2.35-h769bd43_9
  libffi             conda-forge/linux-64::libffi-3.2.1-he1b5a44_1007
  libgcc-ng          conda-forge/linux-64::libgcc-ng-9.3.0-h24d8f2e_16
  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-9.3.0-hdf63c60_16
  llvm-openmp        conda-forge/linux-64::llvm-openmp-10.0.1-hc9558a2_0
  mkl-service        conda-forge/linux-64::mkl-service-2.3.0-py38h516909a_0
  mkl_fft            conda-forge/linux-64::mkl_fft-1.1.0-py38hc1659b7_1
  mkl_random         conda-forge/linux-64::mkl_random-1.1.0-py38hb3f55d8_0
  ncurses            conda-forge/linux-64::ncurses-6.2-he1b5a44_1
  numpy              pkgs/main/linux-64::numpy-1.19.1-py38hbc911f0_0
  numpy-base         pkgs/main/linux-64::numpy-base-1.19.1-py38hfa32c7d_0
  openssl            conda-forge/linux-64::openssl-1.1.1h-h516909a_0
  pip                conda-forge/noarch::pip-20.2.3-py_0
  python             conda-forge/linux-64::python-3.8.5-h1103e12_8_cpython
  python_abi         conda-forge/linux-64::python_abi-3.8-1_cp38
  pytide             conda-forge/linux-64::pytide-1.2.2-py38h9de70de_0
  readline           conda-forge/linux-64::readline-8.0-he28a2e2_2
  setuptools         conda-forge/linux-64::setuptools-49.6.0-py38h32f6830_1
  six                conda-forge/noarch::six-1.15.0-pyh9f0ad1d_0
  sqlite             conda-forge/linux-64::sqlite-3.33.0-h4cf870e_0
  tk                 conda-forge/linux-64::tk-8.6.10-hed695b0_0
  wheel              conda-forge/noarch::wheel-0.35.1-pyh9f0ad1d_0
  xz                 conda-forge/linux-64::xz-5.2.5-h516909a_1
  zlib               conda-forge/linux-64::zlib-1.2.11-h516909a_1009

The following packages will be REMOVED:

  intel-openmp-2020.2-254

The following packages will be SUPERSEDED by a higher-priority channel:

  mkl                             pkgs/main::mkl-2020.2-256 --> conda-forge::mkl-2019.5-281

The following packages will be DOWNGRADED:

  mkl-devel                                      2020.2-257 --> 2018.0.1-h470d631_5


Proceed ([y]/n)? y


Downloading and Extracting Packages
mkl_random-1.1.0     | 394 KB    | ################################################################################################################################################################### | 100% 
mkl_fft-1.1.0        | 180 KB    | ################################################################################################################################################################### | 100% 
mkl-service-2.3.0    | 67 KB     | ################################################################################################################################################################### | 100% 
six-1.15.0           | 14 KB     | ################################################################################################################################################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
(my_pytide) ~/Desktop/Git/pangeo-pytide [master|✔]> python3 -m unittest
..INTEL MKL ERROR: /home/jrmet/anaconda3/envs/my_pytide/lib/python3.8/site-packages/pytide/../../../libmkl_avx2.so: undefined symbol: mkl_sparse_optimize_bsr_trsm_i8.
Intel MKL FATAL ERROR: Cannot load libmkl_avx2.so or libmkl_def.so.
```

vs:

```
(my_pytide_2) ~/Desktop/Git/pangeo-pytide [master|✔]> conda install pytide=1.2.2 -c conda-forge
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/jrmet/anaconda3/envs/my_pytide_2

  added / updated specs:
    - pytide=1.2.2


The following NEW packages will be INSTALLED:

  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge
  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-1_llvm
  ca-certificates    conda-forge/linux-64::ca-certificates-2020.6.20-hecda079_0
  certifi            conda-forge/linux-64::certifi-2020.6.20-py38h32f6830_0
  ld_impl_linux-64   conda-forge/linux-64::ld_impl_linux-64-2.35-h769bd43_9
  libblas            conda-forge/linux-64::libblas-3.8.0-17_openblas
  libcblas           conda-forge/linux-64::libcblas-3.8.0-17_openblas
  libffi             conda-forge/linux-64::libffi-3.2.1-he1b5a44_1007
  libgcc-ng          conda-forge/linux-64::libgcc-ng-9.3.0-h24d8f2e_16
  libgfortran-ng     conda-forge/linux-64::libgfortran-ng-7.5.0-hdf63c60_16
  liblapack          conda-forge/linux-64::liblapack-3.8.0-17_openblas
  libopenblas        conda-forge/linux-64::libopenblas-0.3.10-pthreads_hb3c22a3_4
  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-9.3.0-hdf63c60_16
  llvm-openmp        conda-forge/linux-64::llvm-openmp-10.0.1-hc9558a2_0
  mkl                conda-forge/linux-64::mkl-2019.5-281
  ncurses            conda-forge/linux-64::ncurses-6.2-he1b5a44_1
  numpy              conda-forge/linux-64::numpy-1.19.1-py38hbc27379_2
  openssl            conda-forge/linux-64::openssl-1.1.1h-h516909a_0
  pip                conda-forge/noarch::pip-20.2.3-py_0
  python             conda-forge/linux-64::python-3.8.5-h1103e12_8_cpython
  python_abi         conda-forge/linux-64::python_abi-3.8-1_cp38
  pytide             conda-forge/linux-64::pytide-1.2.2-py38h9de70de_0
  readline           conda-forge/linux-64::readline-8.0-he28a2e2_2
  setuptools         conda-forge/linux-64::setuptools-49.6.0-py38h32f6830_1
  sqlite             conda-forge/linux-64::sqlite-3.33.0-h4cf870e_0
  tk                 conda-forge/linux-64::tk-8.6.10-hed695b0_0
  wheel              conda-forge/noarch::wheel-0.35.1-pyh9f0ad1d_0
  xz                 conda-forge/linux-64::xz-5.2.5-h516909a_1
  zlib               conda-forge/linux-64::zlib-1.2.11-h516909a_1009


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
(my_pytide_2) ~/Desktop/Git/pangeo-pytide [master|✔]> python3 -m unittest
..INTEL MKL ERROR: /home/jrmet/anaconda3/envs/my_pytide_2/lib/python3.8/site-packages/pytide/../../../libmkl_avx2.so: undefined symbol: mkl_sparse_optimize_bsr_trsm_i8.
Intel MKL FATAL ERROR: Cannot load libmkl_avx2.so or libmkl_def.so.
```

So in short, I can run pytide using the v 1.2.0 as discussed in #5 , but I do not manage to install pytide v 1.2.2 using conda.

Any idea how to fix this?

I wonder if the solution may be related to what is discussed by user3098840 here:

https://stackoverflow.com/questions/36659453/intel-mkl-fatal-error-cannot-load-libmkl-avx2-so-or-libmkl-def-so/37160780

I tried quite a few of the other solutions there, but with no success.

**Comment by fbriol** [comment]
Please try it with this [version](https://drive.google.com/file/d/1Jquu8Eq90ktLKsTfWU9AOU5JosuUUCtE/view?usp=sharing).

Download the file, then install with conda locally:

```bash
conda install ./pytide-1.2.2-py38h1e42940_1.tar.bz2
```

**Comment by jerabaul29** [comment]
Many thanks for your help.

Mmmh, that does not work either:

```
(base) ~/Downloads$ conda create --name pytide_from_gdrive
(base) ~/Downloads$ conda activate pytide_from_gdrive
(pytide_from_gdrive) ~/Downloads$ conda install ./pytide-1.2.2-py38h1e42940_1.tar.bz2 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
(pytide_from_gdrive) ~/Downloads$ python3
Python 3.8.3 (default, Jul  2 2020, 16:21:59) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pytide
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jrmet/anaconda3/lib/python3.8/site-packages/pytide/__init__.py", line 12, in <module>
    from . import core
ImportError: libomp.so: cannot open shared object file: No such file or directory
>>> exit()

```

**Comment by jerabaul29** [comment]
Ok, using ```conda install -c conda-forge libgomp``` I can fix the libomp error, but then it has to re-install mkl, and I am back to the same problem:

```
(pytide_from_gdrive) ~/Downloads$  conda install -c conda-forge libgomp 
Collecting package metadata (current_repodata.json): done
Solving environment: / 
The environment is inconsistent, please check the package plan carefully
The following packages are causing the inconsistenc- 

  - <unknown>/linux-64::pytide==1.2.2=py38h1e42940_1
done

## Package Plan ##

  environment location: /home/jrmet/anaconda3/envs/pytide_from_gdrive

  added / updated specs:
    - libgomp


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    libgomp-9.3.0              |      h24d8f2e_16         378 KB  conda-forge
    mkl-2020.2                 |              256       213.9 MB  conda-forge
    ------------------------------------------------------------
                                           Total:       214.3 MB

The following NEW packages will be INSTALLED:

  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge
  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-1_llvm
  ca-certificates    conda-forge/linux-64::ca-certificates-2020.6.20-hecda079_0
  certifi            conda-forge/linux-64::certifi-2020.6.20-py38h32f6830_0
  ld_impl_linux-64   conda-forge/linux-64::ld_impl_linux-64-2.35-h769bd43_9
  libblas            conda-forge/linux-64::libblas-3.8.0-17_openblas
  libcblas           conda-forge/linux-64::libcblas-3.8.0-17_openblas
  libffi             conda-forge/linux-64::libffi-3.2.1-he1b5a44_1007
  libgcc-ng          conda-forge/linux-64::libgcc-ng-9.3.0-h24d8f2e_16
  libgfortran-ng     conda-forge/linux-64::libgfortran-ng-7.5.0-hdf63c60_16
  libgomp            conda-forge/linux-64::libgomp-9.3.0-h24d8f2e_16
  liblapack          conda-forge/linux-64::liblapack-3.8.0-17_openblas
  libopenblas        conda-forge/linux-64::libopenblas-0.3.10-pthreads_hb3c22a3_4
  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-9.3.0-hdf63c60_16
  llvm-openmp        conda-forge/linux-64::llvm-openmp-10.0.1-hc9558a2_0
  mkl                conda-forge/linux-64::mkl-2020.2-256
  ncurses            conda-forge/linux-64::ncurses-6.2-he1b5a44_1
  numpy              conda-forge/linux-64::numpy-1.19.1-py38hbc27379_2
  openssl            conda-forge/linux-64::openssl-1.1.1h-h516909a_0
  pip                conda-forge/noarch::pip-20.2.3-py_0
  python             conda-forge/linux-64::python-3.8.5-h1103e12_8_cpython
  python_abi         conda-forge/linux-64::python_abi-3.8-1_cp38
  readline           conda-forge/linux-64::readline-8.0-he28a2e2_2
  setuptools         conda-forge/linux-64::setuptools-49.6.0-py38h32f6830_1
  sqlite             conda-forge/linux-64::sqlite-3.33.0-h4cf870e_0
  tk                 conda-forge/linux-64::tk-8.6.10-hed695b0_0
  wheel              conda-forge/noarch::wheel-0.35.1-pyh9f0ad1d_0
  xz                 conda-forge/linux-64::xz-5.2.5-h516909a_1
  zlib               conda-forge/linux-64::zlib-1.2.11-h516909a_1009


Proceed ([y]/n)? y


Downloading and Extracting Packages
mkl-2020.2           | 213.9 MB  | ############################################################################# | 100% 
libgomp-9.3.0        | 378 KB    | ############################################################################# | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
(pytide_from_gdrive) ~/Desktop/Git/pangeo-pytide [add-example L|✔]> python3 -m unittest
..INTEL MKL ERROR: /home/jrmet/anaconda3/envs/pytide_from_gdrive/lib/python3.8/site-packages/pytide/../../../libmkl_avx2.so: undefined symbol: mkl_sparse_optimize_bsr_trsm_i8.
Intel MKL FATAL ERROR: Cannot load libmkl_avx2.so or libmkl_def.so.
(pytide_from_gdrive) ~/Desktop/Git/pangeo-pytide [add-example L|✔]> 

```

**Comment by jerabaul29** [comment]
Do you think there is a way to build a docker / singularity container maybe? ^^

**Comment by fbriol** [comment]
The missing package is `llvm-openmp`. Try to do this
```bash
(base) $ conda create -n dummy
$ conda deactivate
$ conda activate dummy
(dummy) $ conda install pytide-1.2.2-py38h1e42940_1.tar.bz2 # This is the downloaded file
(dummy) $ conda install python=3.8 -c conda-forge
Collecting package metadata (current_repodata.json): done
Solving environment: | 
The environment is inconsistent, please check the package plan carefully
The following packages are causing the inconsistency:

  - file:///home/fbriol/Documents/workspace/pytide-feedstock/build_artifacts/linux-64::pytide==1.2.2=py38h1e42940_1
done

## Package Plan ##

  environment location: /home/fbriol/anaconda3/envs/dummy

  added / updated specs:
    - python=3.8


The following NEW packages will be INSTALLED:

  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge
  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-1_llvm
  ca-certificates    conda-forge/linux-64::ca-certificates-2020.6.20-hecda079_0
  certifi            conda-forge/linux-64::certifi-2020.6.20-py38h32f6830_0
  ld_impl_linux-64   conda-forge/linux-64::ld_impl_linux-64-2.35-h769bd43_9
  libblas            conda-forge/linux-64::libblas-3.8.0-17_openblas
  libcblas           conda-forge/linux-64::libcblas-3.8.0-17_openblas
  libffi             conda-forge/linux-64::libffi-3.2.1-he1b5a44_1007
  libgcc-ng          conda-forge/linux-64::libgcc-ng-9.3.0-h24d8f2e_16
  libgfortran-ng     conda-forge/linux-64::libgfortran-ng-7.5.0-hdf63c60_16
  liblapack          conda-forge/linux-64::liblapack-3.8.0-17_openblas
  libopenblas        conda-forge/linux-64::libopenblas-0.3.10-pthreads_hb3c22a3_4
  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-9.3.0-hdf63c60_16
  llvm-openmp        conda-forge/linux-64::llvm-openmp-10.0.1-hc9558a2_0
  mkl                conda-forge/linux-64::mkl-2020.2-256
  ncurses            conda-forge/linux-64::ncurses-6.2-he1b5a44_1
  numpy              conda-forge/linux-64::numpy-1.19.1-py38hbc27379_2
  openssl            conda-forge/linux-64::openssl-1.1.1h-h516909a_0
  pip                conda-forge/noarch::pip-20.2.3-py_0
  python             conda-forge/linux-64::python-3.8.5-h1103e12_8_cpython
  python_abi         conda-forge/linux-64::python_abi-3.8-1_cp38
  readline           conda-forge/linux-64::readline-8.0-he28a2e2_2
  setuptools         conda-forge/linux-64::setuptools-49.6.0-py38h32f6830_1
  sqlite             conda-forge/linux-64::sqlite-3.33.0-h4cf870e_0
  tk                 conda-forge/linux-64::tk-8.6.10-hed695b0_0
  wheel              conda-forge/noarch::wheel-0.35.1-pyh9f0ad1d_0
  xz                 conda-forge/linux-64::xz-5.2.5-h516909a_1
  zlib               conda-forge/linux-64::zlib-1.2.11-h516909a_1009


Proceed ([y]/n)? 

Preparing transaction: done
Verifying transaction: done
Executing transaction: done

(dummy) $ python
Python 3.8.5 | packaged by conda-forge | (default, Sep 16 2020, 18:01:20) 
[GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pytide
>>> pytide
<module 'pytide' from '/home/fbriol/anaconda3/envs/dummy/lib/python3.8/site-packages/pytide/__init__.py'>
>>> 
```

**Comment by jerabaul29** [comment]
It works, many thanks :) This is quite weird, as looking at my previous post:

```
The following NEW packages will be INSTALLED:
  llvm-openmp        conda-forge/linux-64::llvm-openmp-10.0.1-hc9558a2_0
```

So llvm-openmp was well being installed, and in the same version if I am right. Very strange. Do you have some ideas of why this was problematic before?

Do you think that there is a way that you can update the readme / documentation to help the reader?

**Comment by fbriol** [comment]
No, it's the job of conda. The problem is that we used a local file for the test. In this case, conda does not check the dependencies. It does it during an update, or in our case, during the installation of a package.
I'll dump the changes into conda-forge, you'll be able to update from conda tomorrow morning.

**Comment by jerabaul29** [comment]
Ok, sounds good.

Then I will try again tomorrow when you have updated the conda-forge sources, just let me know and I will test and confirm if things work well :) .

Your package does a really excellent work, it actually is perfectly on par or even slightly beats some of the official tide tables I have access to on several coastal stations in UK / Norway / Denmark :) . Many thanks again for your help!

**Comment by fbriol** [comment]
The file is on conda forge (build number 1). Did it solve your problems?

**Comment by jerabaul29** [comment]
Many thanks for your help.

Unfortunately, no :(

```
(base) ~/Desktop/Current/pangeo-pytide [master|✔]> conda create --name pytide_new_build
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/jrmet/anaconda3/envs/pytide_new_build



Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate pytide_new_build
#
# To deactivate an active environment, use
#
#     $ conda deactivate

(base) ~/Desktop/Current/pangeo-pytide [master|✔]> conda activate pytide_new_build
(pytide_new_build) ~/Desktop/Current/pangeo-pytide [master|✔]> conda install mkl-devel
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/jrmet/anaconda3/envs/pytide_new_build

  added / updated specs:
    - mkl-devel


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    mkl-2020.2                 |              256       138.3 MB
    ------------------------------------------------------------
                                           Total:       138.3 MB

The following NEW packages will be INSTALLED:

  blas               pkgs/main/linux-64::blas-1.0-mkl
  intel-openmp       pkgs/main/linux-64::intel-openmp-2020.2-254
  mkl                pkgs/main/linux-64::mkl-2020.2-256
  mkl-devel          pkgs/main/linux-64::mkl-devel-2020.2-257
  mkl-include        pkgs/main/linux-64::mkl-include-2020.2-256


Proceed ([y]/n)? y


Downloading and Extracting Packages
mkl-2020.2           | 138.3 MB  | ############################################################################# | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
(pytide_new_build) ~/Desktop/Current/pangeo-pytide [master|✔]> conda install pytide -c conda-forge
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/jrmet/anaconda3/envs/pytide_new_build

  added / updated specs:
    - pytide


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    _openmp_mutex-4.5          |            1_gnu          22 KB  conda-forge
    mkl-service-2.3.0          |   py38h1e0a361_1          56 KB  conda-forge
    mkl_fft-1.2.0              |   py38hab2c0dc_1         172 KB  conda-forge
    mkl_random-1.2.0           |   py38hc5bc63f_1         333 KB  conda-forge
    ------------------------------------------------------------
                                           Total:         584 KB

The following NEW packages will be INSTALLED:

  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge
  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-1_gnu
  ca-certificates    conda-forge/linux-64::ca-certificates-2020.6.20-hecda079_0
  certifi            conda-forge/linux-64::certifi-2020.6.20-py38h32f6830_0
  ld_impl_linux-64   conda-forge/linux-64::ld_impl_linux-64-2.35-h769bd43_9
  libffi             conda-forge/linux-64::libffi-3.2.1-he1b5a44_1007
  libgcc-ng          conda-forge/linux-64::libgcc-ng-9.3.0-h24d8f2e_16
  libgomp            conda-forge/linux-64::libgomp-9.3.0-h24d8f2e_16
  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-9.3.0-hdf63c60_16
  llvm-openmp        conda-forge/linux-64::llvm-openmp-10.0.1-hc9558a2_0
  mkl-service        conda-forge/linux-64::mkl-service-2.3.0-py38h1e0a361_1
  mkl_fft            conda-forge/linux-64::mkl_fft-1.2.0-py38hab2c0dc_1
  mkl_random         conda-forge/linux-64::mkl_random-1.2.0-py38hc5bc63f_1
  ncurses            conda-forge/linux-64::ncurses-6.2-he1b5a44_1
  numpy              pkgs/main/linux-64::numpy-1.19.1-py38hbc911f0_0
  numpy-base         pkgs/main/linux-64::numpy-base-1.19.1-py38hfa32c7d_0
  openssl            conda-forge/linux-64::openssl-1.1.1h-h516909a_0
  pip                conda-forge/noarch::pip-20.2.3-py_0
  python             conda-forge/linux-64::python-3.8.5-h1103e12_8_cpython
  python_abi         conda-forge/linux-64::python_abi-3.8-1_cp38
  pytide             conda-forge/linux-64::pytide-1.2.2-py38h1e42940_1
  readline           conda-forge/linux-64::readline-8.0-he28a2e2_2
  setuptools         conda-forge/linux-64::setuptools-49.6.0-py38h32f6830_1
  six                conda-forge/noarch::six-1.15.0-pyh9f0ad1d_0
  sqlite             conda-forge/linux-64::sqlite-3.33.0-h4cf870e_0
  tk                 conda-forge/linux-64::tk-8.6.10-hed695b0_0
  wheel              conda-forge/noarch::wheel-0.35.1-pyh9f0ad1d_0
  xz                 conda-forge/linux-64::xz-5.2.5-h516909a_1
  zlib               conda-forge/linux-64::zlib-1.2.11-h516909a_1009


Proceed ([y]/n)? y


Downloading and Extracting Packages
mkl-service-2.3.0    | 56 KB     | ####################################################################################################################### | 100% 
_openmp_mutex-4.5    | 22 KB     | ####################################################################################################################### | 100% 
mkl_random-1.2.0     | 333 KB    | ####################################################################################################################### | 100% 
mkl_fft-1.2.0        | 172 KB    | ####################################################################################################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
(pytide_new_build) ~/Desktop/Current/pangeo-pytide [master|✔]> git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
(pytide_new_build) ~/Desktop/Current/pangeo-pytide [master|✔]> python3 -m unittest
..INTEL MKL ERROR: /home/jrmet/anaconda3/envs/pytide_new_build/lib/python3.8/site-packages/pytide/../../../libmkl_avx2.so: undefined symbol: mkl_sparse_optimize_bsr_trsm_i8.
Intel MKL FATAL ERROR: Cannot load libmkl_avx2.so or libmkl_def.so.
```

**Comment by jerabaul29** [comment]
What I am doing now is that I have my own fork, on which I have suppressed all MKL stuff (I do not think it was making a big difference for me anyways), and then it is easy for me to build from sources:

https://github.com/jerabaul29/pangeo-pytide/tree/add-example

If you manage to get the conda install to work for all users I think it would of course be best though :) .

**Comment by fbriol** [comment]
This should be resolved in the new version 1.2.3. With conda-forge, MKL is used as a single dynamic library.

**Comment by jerabaul29** [comment]
I can confirm that I can install now, many thanks :) 

```
(base) ~$ conda install pytide -c conda-forge
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.8.3
  latest version: 4.8.5

Please update conda by running

    $ conda update -n base -c defaults conda



## Package Plan ##

  environment location: /home/jrlab/anaconda3

  added / updated specs:
    - pytide


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    conda-4.8.5                |   py38h32f6830_1         3.1 MB  conda-forge
    python_abi-3.8             |           1_cp38           4 KB  conda-forge
    pytide-1.2.0               |   py38h9de70de_0         158 KB  conda-forge
    ------------------------------------------------------------
                                           Total:         3.2 MB

The following NEW packages will be INSTALLED:

  python_abi         conda-forge/linux-64::python_abi-3.8-1_cp38
  pytide             conda-forge/linux-64::pytide-1.2.0-py38h9de70de_0

The following packages will be UPDATED:

  conda                       pkgs/main::conda-4.8.3-py38_0 --> conda-forge::conda-4.8.5-py38h32f6830_1


Proceed ([y]/n)? y


Downloading and Extracting Packages
conda-4.8.5          | 3.1 MB    | ############################################################################# | 100% 
python_abi-3.8       | 4 KB      | ############################################################################# | 100% 
pytide-1.2.0         | 158 KB    | ############################################################################# | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
(base) ~$ cd Desktop/Git/pangeo-pytide/
(base) ~/Desktop/Git/pangeo-pytide [master|✔]> python3 -m unittest
EE
======================================================================
ERROR: tests.test_core (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.test_core
Traceback (most recent call last):
  File "/home/jrlab/anaconda3/lib/python3.8/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/home/jrlab/anaconda3/lib/python3.8/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/home/jrlab/Desktop/Git/pangeo-pytide/tests/test_core.py", line 9, in <module>
    import netCDF4
ModuleNotFoundError: No module named 'netCDF4'


======================================================================
ERROR: tests.test_pytide (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.test_pytide
Traceback (most recent call last):
  File "/home/jrlab/anaconda3/lib/python3.8/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/home/jrlab/anaconda3/lib/python3.8/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/home/jrlab/Desktop/Git/pangeo-pytide/tests/test_pytide.py", line 9, in <module>
    import netCDF4
ModuleNotFoundError: No module named 'netCDF4'


----------------------------------------------------------------------
Ran 2 tests in 0.000s

FAILED (errors=2)
(base) ~/Desktop/Git/pangeo-pytide [master|✔]> pip3 install netCDF4
Collecting netCDF4
  Downloading netCDF4-1.5.4-cp38-cp38-manylinux1_x86_64.whl (4.3 MB)
     |████████████████████████████████| 4.3 MB 3.1 MB/s 
Requirement already satisfied: numpy>=1.9 in /home/jrlab/anaconda3/lib/python3.8/site-packages (from netCDF4) (1.18.5)
Collecting cftime
  Downloading cftime-1.2.1-cp38-cp38-manylinux1_x86_64.whl (271 kB)
     |████████████████████████████████| 271 kB 6.1 MB/s 
Installing collected packages: cftime, netCDF4
Successfully installed cftime-1.2.1 netCDF4-1.5.4
(base) ~/Desktop/Git/pangeo-pytide [master|✔]> python3 -m unittest
....E.......
======================================================================
ERROR: test_pickle (tests.test_core.WaveTable)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jrlab/Desktop/Git/pangeo-pytide/tests/test_core.py", line 85, in test_pickle
    unfrozen = pickle.loads(pickle.dumps(wt))
TypeError: cannot pickle 'pytide.core.WaveTable' object

----------------------------------------------------------------------
Ran 12 tests in 30.826s

FAILED (errors=1)
```

Two small things:

- do you think it may be helpful to ship netCDF4 in the package?
- there is still a test failing, not sure why. The strange thing is I do not get this pickling error on my branch.

**Comment by fbriol** [comment]
NetCDF4 is not a dependency to use the library. It is a dependency for test execution. That's why this library is installed during test execution.
The version you have installed is 1.2.0 when you should have installed 1.2.3. The pickle problem no longer exists in this version.
I check your code at the end of the week.

**Comment by jerabaul29** [comment]
Ok, sounds good, thanks!

My code is quite ugly, I will need to re-factor heavily. I also have a few "broken" imports that will only work on my machine that I would fix before a pull request. Basically you should tell me if you think this class is useful:

https://github.com/jerabaul29/pangeo-pytide/blob/b31afeb2c578203c91883ef761e2998198894fbf/src/pytide/__init__.py#L220-L358

The idea is to simplify user API to:

https://github.com/jerabaul29/pangeo-pytide/blob/b31afeb2c578203c91883ef761e2998198894fbf/examples/example_pytide_analyzer.py#L62-L70

**Comment by jerabaul29** [comment]
I am not sure of why an old version of the package was installed by default, that's strange. May be because of the non up to date conda version? I will try again tonight.
